runtime/lua/vim/treesitter/health.lua

```diff
@@ -64,6 +64,53 @@ function M.check()
       )
     end
   end
+
+  health.start('Treesitter queries')
+  local query_files = vim.api.nvim_get_runtime_file('queries/**/*.scm', true)
+  ---@class QueryEntry
+  ---@field lang string
+  ---@field type string
+  ---@field path string
+  ---@field index integer
+  local queries_by_lang = {} ---@type table<string, QueryEntry[]>
+  for i, query_file in ipairs(query_files) do
+    local lang, query_type = query_file:match('queries/([^/]+)/([^/]+)%.scm$')
+    if lang and query_type then
+      if not queries_by_lang[lang] then
+        queries_by_lang[lang] = {}
+      end
+      table.insert(queries_by_lang[lang], {
+        lang = lang,
+        type = query_type,
+        path = query_file,
```

Comment:
This list is going to be very unwieldy; I would recommend stripping the filename  (since it's redundant with the leading `type:`).